From cornelia.huck@de.ibm.com Wed Jul 25 00:58:00 2007
From: Cornelia Huck <cornelia.huck@de.ibm.com>
Date: Wed, 25 Jul 2007 09:58:08 +0200
Subject: driver core: revert "device" link creation check
To: "Kay Sievers" <kay.sievers@vrfy.org>
Cc: "Simon Arlott" <simon@fire.lp0.eu>, "Greg KH" <gregkh@suse.de>, "Linux  Kernel Mailing List" <linux-kernel@vger.kernel.org>
Message-ID: <20070725095808.09b2cc2a@gondolin.boeblingen.de.ibm.com>

From: Cornelia Huck <cornelia.huck@de.ibm.com>

driver core: revert "device" link creation check

Commit 2ee97caf0a6602f749ddbfdb1449e383e1212707 introduced an extra
check on when to create the "device" symlink. Unfortunately, this
breaks input, so let's revert to the old behaviour.

Signed-off-by: Cornelia Huck <cornelia.huck@de.ibm.com>
Acked-by: Kay Sievers <kay.sievers@vrfy.org>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>


---
 drivers/base/core.c |    3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

--- a/drivers/base/core.c
+++ b/drivers/base/core.c
@@ -678,8 +678,7 @@ static int device_add_class_symlinks(str
 		if (error)
 			goto out_subsys;
 	}
-	/* only bus-device parents get a "device"-link */
-	if (dev->parent && dev->parent->bus) {
+	if (dev->parent) {
 		error = sysfs_create_link(&dev->kobj, &dev->parent->kobj,
 					  "device");
 		if (error)
